src/acl/Acl.cc

```diff
@@ -269,21 +281,12 @@ ACL::ParseAclLine(ConfigParser &parser, ACL ** head)
         new_acl = 0;
     }
 
-    /*
-     * Here we set AclMatchedName in case we need to use it in a
-     * warning message in aclDomainCompare().
-     */
-    AclMatchedName = A->name;   /* ugly */
-
-    A->parseFlags();
-
-    /*split the function here */
-    A->parse();
-
-    /*
-     * Clear AclMatchedName from our temporary hack
-     */
-    AclMatchedName = nullptr;  /* ugly */
+    CallContextCreator([A] {
```

Comment:
> Eduard: this adjustment only slightly reduced the noise - leaving out 5 other duplicated context entries. I guess we need to throw instead self_destruct() ...

Please do. I think context reporting noise gives us sufficient grounds for placing that improvement in this PR. The fatal ERROR text will go into the TextException message, of course.

> and CallService() instead of CallContextCreator() in Acl.cc.

Calling CallService() here feels wrong. We do want to associate the exception with the "ACL [parsing service]" and CallService() has the opposite effect.

> With these changes, CodeContext will be reset back to nil automatically before the caught exception leads to another self_destruct().

I am not sure this PR _should_ solve some of the problems you are describing. However, I am OK with trying to solve them: AFAICT, parseOneConfigFile() should call parse_line() using a CallContextCreator() wrapper. I would leave sslBumpCfgRr::finalizeConfig() alone because that "temporary" code should be removed instead.

Reply:
> we need to throw instead self_destruct() 

Done.

> parseOneConfigFile() should call parse_line() using a CallContextCreator() wrapper

Done.

> I would leave sslBumpCfgRr::finalizeConfig()

Ok, we would probably need some comment explaining that we do not adjust this parse_line() because this code is temporary and not maintained. However, there is another parse_line() in the generated default_all(). I think we do not need code context for our generated code (since we trust its validity). But I had to add a temporary wrapper to default_all() to drop the context back to nil after the call (we could just reset() it instead).